File: internal/core/ports/bot.go
package ports

type BotService interface {
	HandlePing() string
	HandlePassenger(chatID int64) string
	AddPassenger(name string, chatID int64) error
	IsWaitingForPassengerName(chatID int64) bool
	ClearWaitingStatus(chatID int64)
	GetPassengerList(chatID int64) (string, error)
	HandleDriver(chatID int64) string
	AddDriver(name string, chatID int64) error
	GetDriverList(chatID int64) (string, error)
	IsWaitingForDriverName(chatID int64) bool
	ProcessDeparture(driverName string, passengers []string, chatID int64) (string, error)
	ProcessReturn(driverName string, passengers []string, chatID int64) (string, error)
	GetTodayReport(chatID int64) (string, error)
	GetReportByDate(chatID int64, date string) (string, error)
}

File: src/core/ports/bot.ts
export interface BotService {
    handlePing(): string;
    handlePassenger(chatId: number): string;
    addPassenger(name: string, chatId: number): Promise<void>;
    isWaitingForPassengerName(chatId: number): boolean;
    clearWaitingStatus(chatId: number): void;
    getPassengerList(chatId: number): Promise<string>;
    handleDriver(chatId: number): string;
    addDriver(name: string, chatId: number): Promise<void>;
    getDriverList(chatId: number): Promise<string>;
    isWaitingForDriverName(chatId: number): boolean;
    processDeparture(driverName: string, passengers: string[], chatId: number): Promise<string>;
    processReturn(driverName: string, passengers: string[], chatId: number): Promise<string>;
    getTodayReport(chatId: number): Promise<string>;
    getReportByDate(chatId: number, date: string): Promise<string>;
    parseAndProcessDeparture(text: string, chatId: number): Promise<string>;
    parseAndProcessReturn(text: string, chatId: number): Promise<string>;
}
